source/common/http/filter_manager.cc

```diff
@@ -922,9 +959,9 @@ void DownstreamFilterManager::sendLocalReply(
 
   // Stop filter chain iteration if local reply was sent while filter decoding or encoding callbacks
   // are running.
-  if (state_.filter_call_state_ & FilterCallState::IsDecodingMask) {
-    state_.decoder_filter_chain_aborted_ = true;
-  } else if (state_.filter_call_state_ & FilterCallState::IsEncodingMask) {
+  // Local reply always stops decoder filter chain.
```

Comment:
this looks like a functional change. Should it be behind this or a separate runtime guard?  maybe split out and land ahead?

Reply:
Made it flag protected. I will create separate PR to make it so that decoder filter is always aborted on local reply.